frontend/public/src/containers/HeaderApp_test.js

```diff
@@ -18,16 +18,46 @@ describe("Top-level HeaderApp", () => {
       notificationsApi,
       "removeStoredUserMessage"
     )
-    renderPage = helper.configureHOCRenderer(HeaderApp, InnerHeaderApp, {}, {})
+    renderPage = helper.configureMountRenderer(
+      HeaderApp,
+      InnerHeaderApp,
+      {},
+      {}
+    )
   })
 
   afterEach(() => {
     helper.cleanup()
   })
 
-  it("fetches user data on load and initially renders an empty element", async () => {
+  /* Because mount will automatically call componentDidMount, I'm simulating a null response from the server
+   * to test the loading state.  Until the component gets a positive response, it will continue to not render the header.
+   * This includes it receiving no parseable response.  I do have a new test after to show what a positive response does.
+   */
```

Comment:
This, like hte above comment, is largely because Enzyme is not using React as the renderer, but another third party renderer which has slightly different behavior